packaging/installer/methods/source.md

```diff
@@ -0,0 +1,289 @@
+
+#### JSON-C
+
+Netdata requires the use of JSON-C for JSON parsing when using Netdata
+Cloud. Netdata is able to use a system-provided copy of JSON-C, but
+some systems may not provide it. If your system does not provide JSON-C,
+you can do the following to prepare a copy for the build system:
+
+1. Verify the tag that Netdata expects to be used by checking the contents
+   of `packaging/jsonc.version` in your Netdata sources.
+2. Obtain the sources for that version by either:
+    * Navigating to https://github.com/json-c/json-c and donwloading
+      and unpacking the source code archive for that release.
+    * Cloning the repository with `git` and checking out the required tag.
+3. Prepare the JSON-C sources by running `cmake -DBUILD_SHARED_LIBS=OFF .`
+   in the JSON-C source directory.
+4. Build JSON-C by running `make` in the JSON-C source directory.
+5. In the Netdata source directory, create a directory called
+   `externaldeps/jsonc`.
+6. Copy `libjson-c.a` fro the JSON-C source directory to
+   `externaldeps/jsonc/libjson-c.a` in the Netdata source tree.
+7. Copy all of the header files (`*.h`) from the JSON-C source directory
+   to `externaldeps/jsonc/json-c` in the Netdata source tree.
+
+## Building Netdata
+
+Once the source tree has been prepared, Netdata is ready to be configured
+and built. Netdata currently uses GNU autotools as it's primary build
+system. To build Netdata this way:
+
+1. Run `autoreconf -ivf` in the Netdata source tree.
+2. Run `./configure` in the Netdata source tree.
+3. Run `make` in the Netdata source tree.
+
+### Configure Options
+
+Netdata provides a number of build time configure options. This section
+lists some of the ones you are most likely to need:
+
+* `--prefix`: Specify the prefix under which Netdata will be installed.
+* `--with-webdir`: Specify a path relative to the prefix in which to
+  install the web UI files.
+* `--disable-cloud`: Disables all Netdata Cloud functionality for
+  this build.
+
+### Using Clang
+
+Netdata is primarily developed using GCC, but in most cases we also
+build just fine using Clang. Under some build configurations of Clang
+itself, you may see build failures with the linker reporting errors
+about `nonrepresentable section on output`. We currently do not have a
+conclusive fix for this issue (the obvious fix leads to other issues which
+we haven't been able to fix yet), and unfortunately the only workaround
+is to use a different build of Clang or to use GCC.
+
+### Linking Errors Relating to OpenSSL
+
+Netdata's build system currently does not reliably support building
+on systems which have multiple ABI incompatible versions of OpenSSL
+installed. In such situations, you may encounter linking errors due to
+Netdata trying to build against headers for one version but link to a
+different version.
+
+## Additional Components
+
+A full featured install of Netdata requires some additional components
```

Comment:
Such as?

Reply:
I would think this is relatively well covered in the subsequent sections under this section. If you feel that we need more elaboration here, I can do that, but it will make any future additions or removals a bit more invasive.